### PR TITLE
fix: PageNumberPaginator not reset when iterating through multiple pa…

### LINKED ISSUE #1921: PageNumberPaginator not working with nested resources
### dlt version

1.1.0

### Describe the problem

I'm implementing a connector from a REST API to Snowflake and I found what seems to be a bug

I'm calling a first endpoint /workflows which returns a list of workflows. This endpoint returns a single page.
Then I call a second endpoint /workflows/{workflow_uid}/jobs which returns for each workflow a list of jobs. This endpoint is paginated with a page param.
Let's say I have two workflows (workflow_uid_1 & workflow_uid_2).
dlt starts by doing all these requests:

- `/workflows/workflow_uid_1/jobs?page=0`
- `/workflows/workflow_uid_1/jobs?page=1`
- `/workflows/workflow_uid_1/jobs?page=2`
- `/workflows/workflow_uid_1/jobs?page=3`

until the page is empty, which is the intended behavior.
Then dlt switches to workflow_uid_2 but starts at page 3:

- `/workflows/workflow_uid_2/jobs?page=3`
- ` /workflows/workflow_uid_2/jobs?page=4`

### Expected behavior

The requests made to `/workflows/workflow_uid_2/jobs` should start with `page=0`

### Steps to reproduce

See description

### Operating system

macOS

### Runtime environment

Local

### Python version

3.11

### dlt data source

rest_api

### dlt destination

Snowflake

### Other deployment details

_No response_

### Additional information

The issue can be fixed by adding two lines of code in the `init_request` of `RangePaginator`:

```
def init_request(self, request: Request) -> None:
        self._has_next_page = True
        self.current_value = self.base_index
        if request.params is None:
            request.params = {}

        request.params[self.param_name] = self.current_value
```
This way, the paginator is reset every time it runs through a new parent resource

### PR DESCRIPTION
**Pull request by paul-godhouse**
…rent ressources

### Related Issues

- Fixes #1921 1921
- Closes #...
- Resolves #...

### CODE REVIEW

File: tests/sources/helpers/rest_client/test_paginators.py [hunk 1]
```diff
@@ -380,6 +380,35 @@ def test_update_state(self):
         paginator.update_state(response, data=NON_EMPTY_PAGE)
         assert paginator.has_next_page is False
 
+
```
Comment (burnash) [inline tests/sources/helpers/rest_client/test_paginators.py:383]:
```suggestion
```

File: tests/sources/helpers/rest_client/test_paginators.py [hunk 2]
```diff
@@ -380,6 +380,35 @@ def test_update_state(self):
...
+        paginator.update_request(request)
+
+        assert paginator.current_value == 2
+        assert paginator.has_next_page is False
+
+        paginator.init_request(request)
+        assert paginator.current_value == 1
+        assert paginator.has_next_page is True
+
+
```
Comment (burnash) [inline tests/sources/helpers/rest_client/test_paginators.py:411]:
```suggestion
```

**Review by burnash — APPROVED** [review]
@paul-godhouse thank you very much for finding this and for the fix.

**Comment by paul-godhouse** [comment]
@burnash you're welcome!
Do you have an approximate idea when will be the next package's release including this fix?

**Comment by burnash** [comment]
@paul-godhouse we plan to release an update next week, the exact day is still tbd. This PR is a very important fix so we'll do our best to release asap.

**Comment by burnash** [comment]
@paul-godhouse, we've just released dlt 1.2.0 with this bugfix.

**Pull request merged**